### PR TITLE
Implement UI

### LINKED ISSUE #38: Implement and refine UI
# Explanation

The goal of this issue is to implement the UI namespace and refine the existing implementation to fit our needs.

After this issue the application should be able to be run using the UI namespace and show the user interface defined in the functional specification document.

You may also include UITests for the application.

# Relevant links or documents

You can find the specification of these elements in the SoftwareDesign document

### PR DESCRIPTION
**Pull request by irmasuppes**
Closes #38. 
Icon change during the active recording is in progress

### Configuration
To run UI you should add path to your configuration file in UI project properties debug section as command line argument.

### Known issues: 

- ChannelConsumingException is thrown if you start the recording after toggling it. I don't think I can handle it. Any help is welcome

### CODE REVIEW

**Comment by sjend** [comment]
This should now provide all specified functionality, including changing the icon depending on the recording status. I encountered some issues with WPF DLL verification, which has supposedly been fixed already, but I have been unable to get it to work properly. I unfortunately do not currently have the time to investigate this any further; there should not be any actual affect on the runtime behaviour, but you might see an exceptions being thrown in the debugger. If you can, please look into this.

**Comment by insightmind** [comment]
The UI application should now work correctly. Please review thoroughly.

File: application/Core/MORR/Data/IntermediateFormat/Json/JsonIntermediateFormatSerializer.cs [hunk 1]
```diff
@@ -51,6 +51,16 @@ private void LinkAllQueues()
...
+        private void UnlinkOnClose()
+        {
+            var canClose = EventQueues.All(queue => queue.IsClosed);
+
+            if (canClose)
+            {
+                Close();
+            }
+        }
+
```
Comment (sjend) [inline application/Core/MORR/Data/IntermediateFormat/Json/JsonIntermediateFormatSerializer.cs]: Why is this necessary @insightmind? The class has a countdown event that automatically calls `Close` once all event queues have exited; this is specifically implemented in lines 93 - 96. This method here will not actually close the producer reliable if the event queues aren't already closed once this is called, which is not really guaranteed.

Comment (insightmind) [reply application/Core/MORR/Data/IntermediateFormat/Json/JsonIntermediateFormatSerializer.cs]: Sorry this is was temporary code while I was trying to figure out why it did not work. Will remove it asap.

**Comment by sjend** [comment]
I have fixed a problem when trying to use the `GraphicsCapturePicker`. This PR now also closes #118.
